### ISSUE TITLE
Breaking changes documentation

### ISSUE THREAD

**Issue by sallgeud** [elastic/elasticsearch-net#1844]
I'm having a fairly painful time trying to find breaking changes between 1.0 and 2.0.

We had been making use of FilterContainer (which is gone). Now using the SearchDescriptor.Query, we're finding that .And, .Not and others are deprecated, but their alternatives are not documented in code.

**Comment by CocaColaBear** [comment]
We use _QueryContainer_ instead of _FilterContainer_
For example:
```
// ...
var queryContainer = new QueryContainerDescriptor<Model>().Bool(b => ...);
// add other bool query instead .And
queryContainer &= new QueryContainerDescriptor<Model>().Bool(b => ...);
// .. 
searchDescriptor.Query(q => queryContainer);
```

**Comment by sallgeud** [comment]
Thanks... Any chance there's going to be documentation on all of this going out?  I may also have a half-dozen other more complex questions on other new ways of doing things. I'm curious if PostFilter is what we want, since we already filter with a querystring on one set of fields.

**Comment by CocaColaBear** [comment]
I'm also waiting for the full documentation :) I trying to use PostFilter, but it doesnt work for me

**Comment by russcam** [comment]
@sallgeud, @CocaColaBear [this blog post](https://www.elastic.co/blog/ga-release-of-nest-2-0-our-dot-net-client-for-elasticsearch) highlights some of the larger changes in 2.0 and includes links to breaking changes for [Elasticsearch.Net](https://github.com/elastic/elasticsearch-net/blob/master/docs/2.0-breaking-changes/elasticsearch-net-breaking-changes.md) and [NEST](https://github.com/elastic/elasticsearch-net/blob/master/docs/2.0-breaking-changes/nest-breaking-changes.md).

We are working on documentation for 2.0 that will be available under [the clients documentation on elastic.co](https://www.elastic.co/guide/en/elasticsearch/client/index.html). [Here's the documentation](https://github.com/elastic/elasticsearch-net/tree/feature/documentation/docs/asciidoc) that we have so far.

@sallgeud [filters and queries merged in Elasticsearch 2.0](https://www.elastic.co/blog/better-query-execution-coming-elasticsearch-2-0) and NEST 2.0 reflects this change by using `QueryContainer` in places where you would have used `FilterContainer` in NEST 1.x. In addition, where you were using `.And()` and `.Not()`, you should be using `bool` query `.Should()` and `MustNot()`, respectively.

@CocaColaBear feel free to open another issue with what is not working for you with PostFilter :smile:
